zcash_primitives/src/zip32/fingerprint.rs

```diff
@@ -0,0 +1,81 @@
+//! Seed Fingerprints according to ZIP 32
+//!
+//! Implements section `Seed Fingerprints` of Shielded Hierarchical Deterministic Wallets (ZIP 32)
+//!
+//! [Section Seed Fingerprints]: https://zips.z.cash/zip-0032#seed-fingerprints
+use blake2b_simd::Params as Blake2bParams;
+
+pub const ZIP32_SEED_FP_PERSONALIZATION: &[u8; 16] = b"Zcash_HD_Seed_FP";
+pub struct SeedFingerprint([u8; 32]);
+
+impl SeedFingerprint {
+    /// Return the seed fingerprint of the wallet as defined in
+    /// <https://zips.z.cash/zip-0032#seed-fingerprints> or None
+    /// if the length of `seed_bytes` is less than 32 or
+    /// greater than 252.
+    pub fn from_seed(seed_bytes: &[u8]) -> Option<SeedFingerprint> {
+        let seed_len = seed_bytes.len();
+
+        if !(32..=252).contains(&seed_len) {
+            return None;
+        }
+
+        let seed_len_byte = if let Ok(len) = u8::try_from(seed_len).map(|l| l) {
+            len
+        } else {
+            return None;
+        };
+
+        match Blake2bParams::new()
+            .hash_length(32)
+            .personal(ZIP32_SEED_FP_PERSONALIZATION)
+            .to_state()
+            .update(&[seed_len_byte])
+            .update(seed_bytes)
+            .finalize()
+            .as_bytes()
+            .try_into()
+        {
+            Ok(blake2b_hash) => Some(SeedFingerprint(blake2b_hash)),
+            Err(_) => None,
+        }
+    }
+
+    /// Returns the fingerprint as a byte array.
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.0
+    }
+}
+
+#[test]
+fn test_seed_fingerprint() {
+    struct TestVector {
+        root_seed: Vec<u8>,
+        fingerprint: Vec<u8>,
+    }
+
+    // From https://github.com/zcash-hackworks/zcash-test-vectors/
```

Comment:
The test vector below is not from `zcash-hackworks/zcash-test-vectors`, so we shouldn't claim it is.
```suggestion
```